### PR TITLE
Error handling for cache

### PR DESCRIPTION
**Pull request by vsafonkin**
Added logic to handle cache errors similarly [`actions/cache`](https://github.com/actions/cache):
https://github.com/actions/cache/blob/main/src/restore.ts#L33

Execution of the action is not interrupted if the cache restore operation returns an error.

### CODE REVIEW

**Review by marko-zivic-93 — APPROVED** [review]

**Review by brcrista — CHANGES_REQUESTED** [review]
This should probably go in the [`restoreCache()` function from the `@actions/cache` toolkit package](https://github.com/actions/toolkit/blob/b5f31bb5a25d129441c294fc81ba7f92f3e978ba/packages/cache/src/cache.ts#L65).  That way everything that uses caching will handle 503s correctly.

**Review by marko-zivic-93 — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Review by panticmilos — APPROVED** [review]

**Comment by brcrista** [comment]
Upstream fix: https://github.com/actions/toolkit/pull/1122

**Comment by vsafonkin** [comment]
@brcrista, please review

**Review by dmitry-shibanov — APPROVED** [review]

**Review by brcrista — CHANGES_REQUESTED** [review]
We just have to bump `@actions/cache` to v3: https://github.com/actions/toolkit/pull/1122

File: src/cache-save.ts [hunk 1]
```diff
@@ -58,18 +58,11 @@ const cachePackages = async () => {
...
-  } catch (error) {
-    if (error.name === cache.ValidationError.name) {
-      throw error;
-    } else if (error.name === cache.ReserveCacheError.name) {
-      core.info(error.message);
-    } else {
-      core.warning(`${error.message}`);
-    }
+  const cacheId = await cache.saveCache(cachePaths, primaryKey);
+  if (cacheId == -1) {
```
Comment (brcrista) [inline src/cache-save.ts:62]:
```suggestion
  if (cacheId === -1) {
```

**Review by brcrista — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Pull request merged**